packages/flutter_tools/lib/src/base/user_messages.dart

```diff
@@ -64,6 +64,11 @@ class UserMessages {
   String get androidUnknownJavaVersion => 'Could not determine java version';
   String androidJavaVersion(String javaVersion) => 'Java version $javaVersion';
   String androidJavaMinimumVersion(String javaVersion) => 'Java version $javaVersion is older than the minimum recommended version of 1.8';
+  String androidJavaMaximumVersion(String javaVersion) =>
+      'Java version $javaVersion is newer than the maximum supported version of the Android SDK tooling which is version 1.8\n'
```

Comment:
The Android Gradle plugin 7.0.0 requires Java 11. Otherwise, the following error is shown:
```
> Android Gradle plugin requires Java 11 to run. You are currently using Java 1.8.
```

The proper fix for this requires to get the Gradle version and then decide what version to suggest in this error message.

Reply:
Here's an example where the tool currently parses the AGP version. https://github.com/flutter/flutter/blob/1bfb7400c8ce546fd19ffd93b458ded583d9ada5/packages/flutter_tools/lib/src/android/gradle_utils.dart#L132 

Ideally, we extract that logic and reuse it for this case.